### ISSUE TITLE
Duplicate relations cause some fields to be null in GQL

### ISSUE THREAD

**Issue by KB1RD** [Jonesus/gatsby-source-directus7#16]
I was creating a schema in my Directus instance and I realized that I messed up one of the relations, so I deleted it. Due to a different bug in Directus, it was *not*, however, deleted from the relations table as one would expect. This created the following entries in my relations table:
```
+----+------------------------------+-------------------+----------------------+---------------------+-------------------+
| id | collection_many              | field_many        | collection_one       | field_one           | junction_field    |
+----+------------------------------+-------------------+----------------------+---------------------+-------------------+
| 17 | blog_tags                    | id                | blog                 | NULL                | NULL              |
| 18 | blog_post_tag_junction       | blog_tags_id      | blog_tags            | blog                | blog_id           |
| 19 | blog_post_tag_junction       | blog_id           | blog                 | tags                | blog_tags_id      |
```
As you can see, `#18` and `#19` are part of the many-to-many relation that I created after I realized my mistake. `#17` is a problematic leftover.

I was able to figure out (via some sketchy console.logs) that GSD7 sees the following relations:
```javascript
[ ...
  { id: 17,
    collection_many: 'blog_tags',
    field_many: 'id',
    collection_one: 'blog',
    field_one: null,
    junction_field: null },
  { id: 18,
    collection_many: 'blog_post_tag_junction',
    field_many: 'blog_tags_id',
    collection_one: 'blog_tags',
    field_one: 'blog',
    junction_field: 'blog_id' },
  { id: 19,
    collection_many: 'blog_post_tag_junction',
    field_many: 'blog_id',
    collection_one: 'blog',
    field_one: 'tags',
    junction_field: 'blog_tags_id' } ]
```
Since GSD7 "saw" `#17` first, it would appear that it gave `#17` control of the `tags` variable in my `blog` GQL query. So I was able to see `blog` from `tags`, but not the other way around. This is because GSD7 rightfully could not find any many-to-one relations between `blog` and `blog_tags`, so it returned nothing. If you examine the console output, you can see this:
```
GSD7: info Found Many-To-One relation: blog -> blog_tags
GSD7: warning Missing One-To-Many relation in 'blog'. The relation
GSD7: warning will be called 'blog_tags' in GraphQL as a best guess.
GSD7: warning Could not find an Many-To-One match in 'blog' for item in 'blog_tags'
GSD7: warning with id '1'. The field value will be left null.
GSD7: warning Could not find an Many-To-One match in 'blog' for item in 'blog_tags'
GSD7: warning with id '2'. The field value will be left null.
GSD7: warning Could not find an Many-To-One match in 'blog' for item in 'blog_tags'
GSD7: warning with id '3'. The field value will be left null.
GSD7: info Found Many-To-Many relation: blog_tags <-> blog
```
...where GSD7 cannot find the relations because the `relations` table contained the invalid `#17`. Deleting this from a SQL prompt solved the problem, however I believe that the error message needs to be more clear. I believe that an additional check should be added for relations with the same two sets that would end up conflicting. Something like this would do:
```
GSD7: warning There are multiple relations in the same direction between 'blog' and 'blog_tags"
```
In short, I think that this is exclusively an issue of having helpful error messages for certain edge cases, so it shouldn't be super high on anyone's priority list. I'll look into doing this and submitting a PR.

**Comment by KB1RD** [comment]
I have an untested fix floating around my computer right now. @Jonesus, after I test the fix, would you rather I committed the fix to the master or develop branch?

**Comment by Jonesus** [comment]
@KB1RD Sounds great! I think the best way would be for you to create a pull request that would then be merged to master for a release :)

**Comment by KB1RD** [comment]
@Jonesus Sorry for the wait time, I'm going to admit that I totally forgot about this for a while... I went to re-test the issue by modifying the database and rebuilding the site and the issue appears to be gone. I did update a few packages, so maybe that changed something, though GSD7 was at the newest version when I last tested. In case this issue pops up again, I have code on my computer which appears to do nothing.
